src/Tasks.UnitTests/AssemblyDependency/Miscellaneous.cs

```diff
@@ -8564,5 +8566,56 @@ private static void GenerateRedistAndProfileXmlLocations(string fullRedistConten
 
             File.WriteAllText(profileRedistList, profileListContents);
         }
+
+        [Fact]
+        public void SDKReferencesAreResolvedWithoutIO()
+        {
+            InitializeRARwithMockEngine(_output, out MockEngine mockEngine, out ResolveAssemblyReference rar);
+
+            string refPath = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString());
```

Comment:
I'm a bit surprised by this since the implementation elsewhere assumes that the filename is the assembly name--having a test case where that's not true doesn't break anything?

Reply:
I have updated the test to use the explicit `AssemblyName` metadatum. Not completely sure I understand the concern. If MSBuild is given the right metadata to activate the "no I/O path", it just trusts what we pass it. In the first version of the PR the assembly name was inferred from file name but, as you pointed out, because it's fragile, I have added a check for `AssemblyName` and made the SDK task pass it to us.